framework/test_selectors/SelectorFactory.cpp

```diff
@@ -188,17 +190,36 @@ struct test *TestrunSelector::testid_to_test(const char *id, bool silent)
     return nullptr;
 }
 
+namespace {
+const char* get_test_list_name_from_cpu_arch()
+{
+    const auto& detected = detect_cpu_arch();
+    if (strcmp(detected.name, unknown_arch.name) == 0) {
+        return nullptr;
+    } else {
+        return detected.codename;
```

Comment:
tolower()? (in simd.conf "short names" are uppercase, I think dictionaries are lowercase..)
Could you attach snippet of generated cpu_features.h?

Reply:
`static const struct X86Architecture x86_architectures[] = {
    { cpu_core2, "Core2", "x86_64" },
    { cpu_westmere, "Westmere", "wsm" },
    { cpu_sandybridge, "Sandy Bridge", "snb" },
    { cpu_silvermont, "Silvermont", "slm" },
    { cpu_ivybridge, "Ivy Bridge", "ivb" },
    { cpu_goldmont, "Goldmont", "glm" },
    { cpu_haswell, "Haswell", "hsw" },
    { cpu_broadwell, "Broadwell", "bdw" },
    { cpu_tremont, "Tremont", "tnt" },
    { cpu_skylake, "Skylake", "skl" },
    { cpu_skylake_avx512, "Skylake (Avx512)", "skx" },
    { cpu_cascadelake, "Cascade Lake", "clx" },
    { cpu_cooperlake, "Cooper Lake", "cpx" },`

Reply:
I used lc() in perl script

Reply:
I've missed it :D